src/Security/Authentication/Certificate/src/CertificateAuthenticationExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authentication;
+
+using Microsoft.AspNetCore.Authentication.Certificate;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods to add Certificate authentication capabilities to an HTTP application pipeline.
+    /// </summary>
+    public static class CertificateAuthenticationAppBuilderExtensions
+    {
+        public static AuthenticationBuilder AddCertificate(this AuthenticationBuilder builder)
+            => builder.AddCertificate(CertificateAuthenticationDefaults.AuthenticationScheme);
+
+        public static AuthenticationBuilder AddCertificate(this AuthenticationBuilder builder, string authenticationScheme)
+            => builder.AddCertificate(authenticationScheme, configureOptions: null);
+
+        public static AuthenticationBuilder AddCertificate(this AuthenticationBuilder builder, Action<CertificateAuthenticationOptions> configureOptions)
+            => builder.AddCertificate(CertificateAuthenticationDefaults.AuthenticationScheme, configureOptions);
```

Comment:
Call the overload that does not set the display name. This auth handler won't work in the normal Identity select-your-auth-type flow, it doesn't support Challenge.

There is one flow where you could allow the user to prompt for a cert:
1) Use HttpSys with lazy cert negotiation enabled (default)
1) The user selects Certificate from the list.
1) The controller calls AuthenticateAsync("Certificate") rather than challenge.
1) The browser prompts for a certificate (on the same request).
1) The controller does whatever signin action it wants.

Reply:
Its only the AddRemoteScheme method that takes a display name, I don't believe this currently is setting a display name so it won't show up in identity flows by default.